### PR TITLE
Fix bug #1782, forbidden chars in ID PATCH v2

### PR DESCRIPTION
**Pull request by crbrox**
Fix bug #1782, forbidden chars in ID PATCH v2

@kzangeli 
@fgalan

### CODE REVIEW

File: src/lib/serviceRoutinesV2/patchEntity.cpp [hunk 1]
```diff
@@ -71,6 +71,12 @@ std::string patchEntity
   eP->id = compV[2];
   eP->type = ciP->uriParam["type"];
 
+  if (forbiddenIdChars(ciP->apiVersion, eP->id.c_str() , NULL))
+  {
+    OrionError oe(SccBadRequest, "invalid character in URI");
+    return oe.render(ciP, "");
+  }
+
```
Comment (kzangeli) [inline src/lib/serviceRoutinesV2/patchEntity.cpp]: This seems like something that should be done for **all** v2 requests (perhaps v1 also),
not only PATCH /v2/entities/<eid> ...
Should the check be on a higher level?
Somewhere in rest.cpp ...

Comment (kzangeli) [inline src/lib/serviceRoutinesV2/patchEntity.cpp]: Ah, it's only about the ID, not the whole path ...

NTC

**Comment by kzangeli** [comment]
LGTM

File: src/lib/serviceRoutinesV2/patchEntity.cpp (hunk 1 above)
Comment (crbrox) [inline src/lib/serviceRoutinesV2/patchEntity.cpp]: I'll check  other 'paths' with ID for this point. If any is found, we can create an issue for all them and solve them in the same PR  ...

File: CHANGES_NEXT_RELEASE [hunk 2]
```diff
@@ -16,3 +16,5 @@
 - Add: type param for POST entity in v2 (Issue #982, #984)
 - Add: support for geo:point type as a way of specifying location attribute in NGSIv2 (Issue #1038)
 - Add: type param for PUT entity in v2 (Issue #988, #992, #1000)
+- Fix: not detecting forbidden chars in ID for PATH v2 (Issue #1782)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: in ID -> in entityID ?
PATH -> PATCH ?

Comment (crbrox) [inline CHANGES_NEXT_RELEASE]: Fixed 297de23

**Comment by fgalan** [comment]
LGTM

**Pull request merged**